models/certs.go

```diff
@@ -32,3 +34,51 @@ type CertInfo struct {
 	ValidTo   string    `json:"ValidTo"`
```

Comment:
### 1. Import Statement

The import statement `import "github.com/hashicorp/terraform-plugin-framework/types"` is added at the top of the file. This is good as it directly imports the required package for the Terraform provider. However, it would be better to group similar imports together (e.g., all standard library imports at the top, followed by third-party imports).

```python
"""
imports
"""
import (
    "github.com/hashicorp/terraform-plugin-framework/types"
    // other imports
)
```

### 2. Code Organization and Comments

The code is well-organized, but it's good to add comments to explain the purpose of each struct, function, and any complex logic.

```python
// CSRConfigModel - CSR generation tfsdk form
// This struct holds the CSR generation form configuration
type CSRConfigModel struct {
    ... // attributes
}
```

### 3. Function Naming and Convention

The function names `NewCSRConfigModel` and `NewCertInfoModel` are descriptive, but it's better to follow the convention of using lowercase letters and underscores to separate words. For example, `newcsrconfigmodel` and `newcertinfomodel`.

```python
func newCSRConfigModel(input CSRConfig) CSRConfigModel {
    ... // code
}

func newCertInfoModel(info CertInfo) CertInfoModel {
    ... // code
}
```

### 4. Error Handling

There's no error handling in the functions `NewCSRConfigModel` and `NewCertInfoModel`. It's essential to add proper error handling to handle potential errors.

```python
func newCSRConfigModel(input CSRConfig) (CSRConfigModel, error) {
    ... // code
}

func newCertInfoModel(info CertInfo) (CertInfoModel, error) {
    ... // code
}
```

### 5. Code Readability

The code is relatively easy to read, but some code smells like duplicated code (e.g., `types.StringValue(input.DistinguishedName)` in both functions) can be refactored to reduce redundancy.

```python
func getStringValue(input string) types.String {
    return types.StringValue(input)
}

func newCSRConfigModel(input CSRConfig) CSRConfigModel {
    ... // code
    DistinguishedName: getStringValue(input.DistinguishedName),
    // more attributes
}

func newCertInfoModel(info CertInfo) CertInfoModel {
    ... // code
    ValidTo: getStringValue(info.ValidTo),
    // more attributes
}
```

Overall, the code is well-structured, but with some improvements, it can be more maintainable, efficient, and readable.